enzy_htp/structure/structure_constraint/api.py

```diff
@@ -1,43 +1,477 @@
-"""Defines the StructureConstraint class.
-This class is one of the data structure of EnzyHTP.
-StructureConstraint stands for a coordinate constrain for the structure, including
-freeze coordinate and geometry constrain (distance, angle, dihedral).
+"""Defines the StructureConstraint abstract base class as well as the inheriting classes that specialize it. 
+StructureConstraints are meant to define flexible, non-package specific relationships that can be translated 
+in between different software packages. Factory functions for constraints are also included. The contained 
+classes are:
+
+    + StructureConstraint: The abstract base class.
+    + CartesianFreeze: Specific atoms will be frozen.
+    + DistanceConstraint: Constraint that specifies a distance.
+    + AngleConstraint: Constraint that specifies an Angle.
+    + DihedralConstraint: Constraint that specifies a dihedral.
+    + ResiduePairConstraint: Specifies a constraint set between two residues in a style similar to Rosetta EnzDes.
```

Comment:
Making them seperate descendant classes is absolutly the right move! In most cases, methods will be specialized for each specific types of constraints. Great job!

Reply:
Thanks!